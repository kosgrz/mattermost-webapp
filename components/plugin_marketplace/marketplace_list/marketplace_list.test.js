// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import MarketplaceList from './marketplace_list';

describe('components/marketplace/marketplace_list', () => {
    const samplePlugin = {
        homepage_url: 'https://github.com/mattermost/mattermost-plugin-nps',
        download_url: 'https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz',
        manifest: {
            id: 'com.mattermost.nps',
            name: 'User Satisfaction Surveys',
            description: 'This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost',
            version: '1.0.3',
            minServerVersion: '5.14.0',
        },
        installed_version: '',
    };

    it('should render with multiple plugins', () => {
        const wrapper = shallow(
            <MarketplaceList
                theme={{centerChannelColor: '#3d3c40'}}
                plugins={[
                    samplePlugin, samplePlugin, samplePlugin, samplePlugin, samplePlugin,
                    samplePlugin, samplePlugin, samplePlugin, samplePlugin, samplePlugin,
                    samplePlugin, samplePlugin, samplePlugin, samplePlugin, samplePlugin,
                    samplePlugin, samplePlugin
                ]}
            />
        );
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.state('page')).toEqual(0);
        expect(wrapper.find('Connect(MarketplaceItem)')).toHaveLength(15);
        expect(wrapper.find('#marketplaceModalNextButton')).toHaveLength(1);
        expect(wrapper.find('#marketplaceModalPreviousButton')).toHaveLength(0);

        wrapper.find('#marketplaceModalNextButton').simulate('click', {preventDefault: jest.fn});

        expect(wrapper.state('page')).toEqual(1);
        expect(wrapper.find('Connect(MarketplaceItem)')).toHaveLength(2);
        expect(wrapper.find('#marketplaceModalNextButtXon')).toHaveLength(0);
        expect(wrapper.find('#marketplaceModalPreviousButton')).toHaveLength(1);
    });

    it('should set page to 0 when list of plugins changed', () => {
        const wrapper = shallow(
            <MarketplaceList
                theme={{centerChannelColor: '#3d3c40'}}
                plugins={[samplePlugin, samplePlugin]}
            />
        );

        wrapper.setState({page: 10});
        wrapper.setProps({plugins: [samplePlugin]});

        expect(wrapper.state('page')).toEqual(0);
    });
});
