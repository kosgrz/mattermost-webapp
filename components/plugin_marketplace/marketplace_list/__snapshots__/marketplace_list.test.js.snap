// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/marketplace/marketplace_list should render with multiple plugins 1`] = `
<div
  className="more-modal__list"
>
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <Connect(MarketplaceItem)
    description="This plugin sends quarterly user satisfaction surveys to gather feedback and help improve Mattermost"
    downloadUrl="https://github.com/mattermost/mattermost-plugin-nps/releases/download/v1.0.3/com.mattermost.nps-1.0.3.tar.gz"
    homepageUrl="https://github.com/mattermost/mattermost-plugin-nps"
    id="com.mattermost.nps"
    installedVersion=""
    isPrepackaged={false}
    key="com.mattermost.nps"
    name="User Satisfaction Surveys"
    version="1.0.3"
  />
  <div
    className="row"
  >
    <div
      className="col-xs-2"
    />
    <div
      className="col-xs-8"
      style={
        Object {
          "color": "rgba(61,60,64,0.6)",
        }
      }
    >
      <FormattedMessage
        defaultMessage="{startCount, number} - {endCount, number} {count, plural, one {plugin} other {plugins}} of {total, number} total"
        id="marketplace_list.countTotalPage"
        values={
          Object {
            "count": 17,
            "endCount": 15,
            "startCount": 1,
            "total": 17,
          }
        }
      />
    </div>
    <div
      className="col-xs-2"
    >
      <button
        className="btn btn-link"
        id="marketplaceModalNextButton"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Next"
          id="more_channels.next"
          values={Object {}}
        />
      </button>
    </div>
  </div>
</div>
`;
